test/extensions/filters/http/ratelimit/ratelimit_integration_test.cc

```diff
@@ -165,13 +165,19 @@ class RatelimitIntegrationTest : public Grpc::VersionedGrpcClientIntegrationPara
 
   void cleanup() {
     if (fake_ratelimit_connection_ != nullptr) {
+      /*
       if (clientType() != Grpc::ClientType::GoogleGrpc) {
         // TODO(htuch) we should document the underlying cause of this difference and/or fix it.
         AssertionResult result = fake_ratelimit_connection_->close();
         RELEASE_ASSERT(result, result.message());
       }
+      */
```

Comment:
IIUC this PR somehow addresses the TODO above; is that right?

Reply:
I don't know, even Harvey didn't remember why this TODO is here.

Reply:
Let's sort this out and clean it up :)

Reply:
Sure, I am still look into it. will clean up later.